src/mlpack/methods/ann/rnn_impl.hpp

```diff
@@ -304,13 +304,10 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
       responseSeq = seqNum;
     }
 
-    if (!deterministic)
```

Comment:
I agree with the documentation changes, but I'm really not sure about this one.  Why should we remove the `if (!deterministic)` here?

Reply:
I think on this https://github.com/mlpack/mlpack/pull/1615/files#diff-0e560b4ae5054e7fe4c029ec9eb0f1a7R275
we set `deterministic` to false.

Reply:
`EvaluateWithGradient` is only being used for training otherwise we could call `Evaluate`.

Reply:
Thanks for the clarification; that makes sense to me.